### PR TITLE
GUVNOR-2338: Copy/Rename modal doesn't disappear when used from Project Explorer links

### PR DESCRIPTION
**Pull request by paulovmr**
Depends on: https://github.com/uberfire/uberfire-extensions/pull/174

### CODE REVIEW

**Comment by paulovmr** [comment]
jenkins retest this

File: jbpm-designer-client/src/main/java/org/jbpm/designer/client/DesignerPresenter.java [hunk 1]
```diff
@@ -247,53 +248,74 @@ public void assetCopyEvent( String uri ) {
...
         } ).get( URIUtil.encode( uri ) );
     }
 
+    private HasBusyIndicatorDefaultErrorCallback getCopyErrorCallback( final CopyPopupView copyPopupView ) {
+        return new HasBusyIndicatorDefaultErrorCallback( baseView ) {
+
+            @Override
+            public boolean error( final Message message,
+                                  final Throwable throwable ) {
+                copyPopupView.hide();
```
Comment (manstis) [inline jbpm-designer-client/src/main/java/org/jbpm/designer/client/DesignerPresenter.java]: For consistency with the ```show()` we should call ```CopyPopup.hide()```

File: jbpm-designer-client/src/main/java/org/jbpm/designer/client/DesignerPresenter.java [hunk 2]
```diff
@@ -334,26 +357,40 @@ public void assetRenameEvent( String uri ) {
...
+        } ).get( URIUtil.encode( uri ) );
+    }
+
+    private HasBusyIndicatorDefaultErrorCallback getRenameErrorCallback( final RenamePopupView renamePopupView ) {
+        return new HasBusyIndicatorDefaultErrorCallback( baseView ) {
+
+            @Override
+            public boolean error( final Message message,
+                                  final Throwable throwable ) {
+                renamePopupView.hide();
```
Comment (manstis) [inline jbpm-designer-client/src/main/java/org/jbpm/designer/client/DesignerPresenter.java]: For consistency with the ```show()` we should call ```RenamePopup.hide()```

File: jbpm-designer-client/src/main/java/org/jbpm/designer/client/DesignerPresenter.java [hunk 3]
```diff
@@ -404,31 +441,33 @@ public void execute() {
...
             }
         };
     }
 
-    private RemoteCallback<Path> getCopySuccessCallback() {
+    private RemoteCallback<Path> getCopySuccessCallback( final CopyPopupView copyPopupView ) {
         return new RemoteCallback<Path>() {
             @Override
             public void callback( final Path path ) {
+                copyPopupView.hide();
```
Comment (manstis) [inline jbpm-designer-client/src/main/java/org/jbpm/designer/client/DesignerPresenter.java]: For consistency with the ```show()` we should call ```CopyPopup.hide()```

File: jbpm-designer-client/src/main/java/org/jbpm/designer/client/DesignerPresenter.java [hunk 4]
```diff
@@ -404,31 +441,33 @@ public void execute() {
...
         };
     }
 
-    private RemoteCallback<Path> getRenameSuccessCallback() {
+    private RemoteCallback<Path> getRenameSuccessCallback( final RenamePopupView renamePopupView ) {
         return new RemoteCallback<Path>() {
 
             @Override
             public void callback( final Path path ) {
+                renamePopupView.hide();
```
Comment (manstis) [inline jbpm-designer-client/src/main/java/org/jbpm/designer/client/DesignerPresenter.java]: For consistency with the ```show()` we should call ```RenamePopup.hide()```

**Comment by manstis** [comment]
Other than the change to use the Presenter's hide() method I think this is much clearer to follow than the flag you had before. Thanks for making the changes... once you switch to using the Presenter I'll OK the merge.

**Comment by manstis** [comment]
+1 to merge.

**Comment by jlindop** [comment]
There are currently two compilation errors in DesignerPresenter:
[ERROR] /home/jeremy/Documents/gitRepositories/droolsjbpm/jbpm-designer/jbpm-designer-client/src/main/java/org/jbpm/designer/client/DesignerPresenter.java:[251,61] error: getDefaultView() has private access in CopyPopup
[ERROR] /home/jeremy/Documents/gitRepositories/droolsjbpm/jbpm-designer/jbpm-designer-client/src/main/java/org/jbpm/designer/client/DesignerPresenter.java:[360,67] error: getDefaultView() has private access in RenamePopup

**Comment by paulovmr** [comment]
Hi @jlindop , it's because this PR depends on https://github.com/uberfire/uberfire-extensions/pull/174

**Comment by jlindop** [comment]
@paulovmr ah yes, my mistake. Apologies!

**Comment by paulovmr** [comment]
@jlindop No problem. :)

**Comment by csadilek** [comment]
master: 42f15f2ba4b604b0f098303cfe8731066f5eaf04

**Pull request closed without merging**